src/modules/control/src/control_node.cpp

```diff
@@ -143,7 +143,8 @@ namespace node
       prev_pos[0] = position(0);
       prev_pos[1] = position(1);
     }*/
-
+    
+    linear_velocity = veh_state_msg.Speed.twist.linear.x;
```

Comment:
I forgot to mention that the Chassis_state message is in the odom/gps frame. This should be changed to:
`linear_velocity =  sqrt(veh_state_msg.Speed.twist.linear.x^2 + veh_state_msg.Speed.twist.linear.y^2);`

Reply:
Fixed